internal/dlv/doc.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package dlv allows to run any binaries with latest version of delve.
+//
+// To make it works:
+//  1. compile your code with
+//     -gcflags "all=-N -l"
+//  2. Use [NewDlvRunner] function to encapsulate your main function.
+//
+// # Tips
+//
+// Don't bring delve in production: use compile tags
+//
+// Refactor your code in such way you can easily call the main workload as a one-liner in `main()` function.
+// Then use compile tag to ensure that the actual main function will wrap the workload in delve only if the debug flag
+// is used at compile time:
+//
+//	go build -gcflags "all=-N -l" -tags debug path/to/my/package
+//
+// It can be done through having two main files in package main, one named `main.go` and one name `main_debug.go`. The
+// former will be compiled only if there is no debug tag, the former only if there is a debug tag. Making mistake with
```

Comment:
"the _latter_ only if there is a debug tag."

Reply:
Done https://github.com/juju/juju/pull/18114/commits/8735fc258e9495c527885ea569ef7368901991d6